app/adapters/request-config.js

```diff
@@ -0,0 +1,25 @@
+import AdapterError from '@ember-data/adapter/error';
+import V3Adapter from 'travis/adapters/v3';
+
+export default V3Adapter.extend({
+  queryRecord(store, type, { id, data }) {
+    const url = `${this.buildURL('repo', id)}/request/config`;
+    return this.ajax(url, 'POST', { data });
+  },
```

Comment:
This looks very much like [`createRecord`](https://api.emberjs.com/ember-data/3.12/classes/RESTAdapter/methods/createRecord?anchor=createRecord). It's doing [literally the same](https://github.com/emberjs/data/blob/v3.12.6/packages/adapter/addon/rest.js#L726-L732). Are there any specific reasons to use `queryRecord` here?

Reply:
hmmm, are you sure? 

it seems the URL would be different, wouldn't it? `this.buildURL(type.modelName, null, snapshot, 'createRecord')` 

and `serializeIntoHash` seems to expect a model to be passed, not an arbitrary JS object https://github.com/emberjs/data/blob/v3.17.0/packages/serializer/addon/rest.js#L667

Reply:
@svenfuchs you may be right about URL and serialization logic, but better approach (and "Ember way") would be to override appropriate API hooks and tweak as necessary. E.g. overriding [`buildURL`](https://api.emberjs.com/ember-data/3.12/classes/RESTAdapter/methods/buildURL?anchor=buildURL) hook would allow you to tweak URL.

Reply:
@artursmirnov i've tried to make this work, but i'm just failing at it. i might be misunderstanding what you are saying about the appropriate API hooks, but my understanding is that `createRecord` expects to be passed an instance of a model, because `serializeIntoHash` does. we don't have such record instance though, and i wouldn't know what to pass here.

respectfully, i'd leave this as is and leave it for a more skilled person to refactor this later, if necessary at all.

Reply:
@svenfuchs okay, let's leave it as is for now 👍 The reason why I pushed so hard on this is that current use of API is incorrect. `queryRecord` is aimed to **fetch** records from server, not to **save** data. This may not only be confusing for other devs and unexpected during debugging, but also may accidentally break after dependencies updates. We would have to refactor it later. I'll put it to my list  ✅ Thanks for trying! :)